Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -788,6 +799,22 @@ private void Update()
                 UpdateRigHandles();
                 Target.transform.hasChanged = false;
             }
+            else if (HasBoundsOverrideChanged())
+            {
+                UpdateBounds();
+                UpdateRigHandles();
+            }
+        }
+        private bool HasBoundsOverrideChanged()
+        {
+            if (boundsOverride == null)
+            {
+                return false;
+            }
+            Bounds curBounds = boundsOverride.bounds;
+            bool result = curBounds != prevBounds;
```

Comment:
Do you know if boundsOverride can really only be updated by changing BoundsOverride, or if there are also pretty common cases where the property itself is changed directly (i.e. play-mode live edits of properties?) I assume that the play mode live edits is a thing but just wanted to confirm.

The reason I ask is, otherwise it seems like the only place where the "bounds is dirty" state could be flipped is in BoundsOverride itself - as such it would seem to be more performant (though by how much, I have no idea) to just set a dirty bool in the setter, and then reset it here (so that you don't need to do a multi compare here on each update tick).

Even if the answer to the first is no, I don't think there's actually any action to be taken here (I don't have enough background on the significant of perf implications, and don't generally believe in making premature 'optimizations')

But just wanted to go through the thought process.

Reply:
boundsOverride is usually modified directly, not through the accessor, for example like this:

```
var newObject = new GameObject();
var bc = newObject.AddComponent<BoxCollider>();
bc.center = new Vector3(.25f, 0, 0);
bc.size = new Vector3(0.162f, 0.1f, 1);
bbox.BoundsOverride = bc;
// This line of code directly modifies bc.size, no way I can think of to set a 'dirty big'
bc.size = new Vector3(0.5f, 0.1f, 1);
```

I don't think this adds much overhead because we already do quite a lot of comparing previous / current state in the Update() loop.